packages/protocol/contracts/L1/libs/LibProvingAlt.sol

```diff
@@ -229,8 +162,8 @@ library LibProvingAlt {
         // A special return value from the top tier prover can signal this
         // contract to return all liveness bond.
         {
-            bool returnLivenessBond = tier.contestBond == 0 && blk.livenessBond > 0
-                && proof.data.length == 32 && bytes32(proof.data) == RETURN_LIVENESS_BOND;
+            bool returnLivenessBond = isGuardian && blk.livenessBond > 0 && proof.data.length == 32
```

Comment:
Lets not assume the top tier must be the guardian prover. When guardian is removed, the top tier may be something else.

Reply:
Rename isGuardian to isTopTier